spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,260 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe(Tapioca::Compilers::Dsl::ActiveRecordTypedStore) do
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecordTypedStore classes") do
+      expect(subject.processable_constants).to(be_empty)
+    end
+
+    it("gather only TypedStore classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+
+        class CustomePost < Post
+        end
+        class Shop < ActiveRecord::Base
+        end
+
+        class User
+        end
+      RUBY
+
+      expect(constants_from(content)).to(eq(["CustomePost", "Post"]))
+    end
+  end
+
+  describe("#decorate") do
+    def rbi_for(content)
+      with_content(content) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    it("generates no definitions if there are no accessors to define") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
```

Comment:
Can we also add another typed_store here in the same class that does:
```
typed_store :properties, accessors: false do
  s.string(:title)
  s.integer(:comment_count)
end
```
to test that case too?